### PR TITLE
SDK/Components - Renamed fileOutputs to unconfigurableOutputPaths in ContainerSpec

### PR DESCRIPTION
**Pull request by Ark-kun**
This feature is deprecated and will only be supported until the existing legacy components are upgraded (See https://github.com/kubeflow/pipelines/pull/580)

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/pipelines/688)

### CODE REVIEW

**Comment by Ark-kun** [comment]
* The feature is current not used by any developer. The only way right now to generate component yaml definitions is to create python components which do not use that feature (and people do not yet know they can output the component file).
* The feature is incompatible with most of the future storage and data management scenarios: Volume-based storage system, Data caching etc, since for those features the output file paths must be set by the system. The volume-based storage system cannot mount volumes to root directory that some of our sample components use ("/output.txt").
* The feature is only needed to support the current version of our sample components and will be removed right after #580 is completed.

This is a trivial change, but I've talked with team members to get the feedback about it.

@gaoning777 Was slightly against the renaming as the feature will be removed anyways and the schema is not fully accepted yet. The counter-argument is that the removal might take a bit more time and at that time the old name remains operational. The final accepted schema will likely either specify the new name or remove it at all. In both cases the old name will go away, so we can do it sooner rather than later.

@rileyjbauer Approved the renaming to discourage the people from using the property. Even if the schema documentation states that the property is deprecated, people can easily miss that.

@IronPan Approved the renaming of the feature (no one using it, the new name is better), but raised concern about strict API versioning. I agree that the feature (with any name) should not make it into the .proto API schema.

@yebrahim Had small concern about someone already using the feature. He agreed though that even in this case removing that feature (name) sooner is better than doing it later.

File: sdk/python/kfp/components/_structures.py [hunk 1]
```diff
@@ -190,15 +190,15 @@ def __init__(self,
 class ContainerSpec(ModelBase):
     '''Describes the container component implementation.'''
     _serialized_names = {
-        'file_outputs': 'fileOutputs', #TODO: rename to something like legacy_unconfigurable_output_paths
+        'unconfigurable_output_paths': 'unconfigurableOutputPaths', #TODO: rename to something like legacy_unconfigurable_output_paths
```
Comment (IronPan) [inline sdk/python/kfp/components/_structures.py]: could you have comments on the reason of this name? also is the todo still needed?

Comment (Ark-kun) [reply sdk/python/kfp/components/_structures.py]: Done.

**Comment by gaoning777** [comment]
My concern is:
1) if no one is using the yaml, why not remove it altogether a little later since this is a small change, which will not delay the feature publishing. Besides, this is not helping anyone assuming no one is using it.
2) if people are already using it, why break their shared yaml by introducing a different name. We should try to keep backward compatible as much as possible.
3) In the same case assuming people are using it, this intermediate solution is causing even more confusion. People might wonder if this is unconfigurable, why not hide it? what on earth is this field, then? etc. 

Above all, I do not really see any benefits except to convey a message that this field will be deprecated however they are not getting warned that the new compiler is not gonna work with their old generated yaml any more.

On the other hand, if everyone approves this change, I do not see too much harm, though.

**Pull request closed without merging**